File: backend/src/main/kotlin/org/icpclive/events/codeforces/api/data/CFContest.java
package org.icpclive.events.codeforces.api.data;

/**
 * @author egor@egork.net
 */
public class CFContest {
    public int id;
    public String name;
    public CFContestType type;
    public CFContestPhase phase;
    public boolean frozen;
    public long durationSeconds;
    public Long startTimeSeconds;
    public Long relativeTimeSeconds;
    public String preparedBy;
    public String websiteUrl;
    public String description;
    public Integer difficulty;
    public String kind;
    public String icpcRegion;
    public String country;
    public String city;
    public String season;

    public enum CFContestType {
        CF, IOI, ICPC
    }

    public enum CFContestPhase {
        BEFORE, CODING, PENDING_SYSTEM_TEST, SYSTEM_TEST, FINISHED
    }
}

File: backend/src/main/kotlin/org/icpclive/cds/codeforces/api/data/CFContest.kt
@file:Suppress("UNUSED")

package org.icpclive.cds.codeforces.api.data

import kotlinx.serialization.Serializable

enum class CFContestType {
    CF, IOI, ICPC
}

enum class CFContestPhase {
    BEFORE, CODING, PENDING_SYSTEM_TEST, SYSTEM_TEST, FINISHED
}

@Serializable
data class CFContest(
    val id: Int,
    val name: String,
    val type: CFContestType,
    val phase: CFContestPhase,
    val frozen: Boolean,
    val durationSeconds: Long? = null,
    val startTimeSeconds: Long? = null,
    val relativeTimeSeconds: Long? = null,
    val preparedBy: String? = null,
    val websiteUrl: String? = null,
    val description: String? = null,
    val difficulty: Int? = null,
    val kind: String? = null,
    val icpcRegion: String? = null,
    val country: String? = null,
    val city: String? = null,
    val season: String? = null,
)